DEVELOPMENT.md

```diff
@@ -212,12 +211,60 @@ to be edited by hand.
 
 3.  Run `yarn run yarn-deduplicate`.
 
-4.  Cross reference your updates with the bazel `WORKSPACE` file and determine if any
-    bazel dependencies should also be updated.
-    * Googlers, for information on how to mirror WORKSPACE dependencies that
-      need to be downloaded, refer to go/tensorboard-tf-mirror.
 
-5.  Rebuild and test TensorBoard to make sure it works:
+4.  Rebuild and test TensorBoard to make sure it works:
     * `rm -rf node_modules; bazel clean --expunge; yarn`
     * `bazel run tensorboard --logdir <your favorite logdir>`
     * `bazel test --test_output=errors tensorboard/webapp/...`
+
+## Updating rules_nodejs
+
+The [bazel](https://bazel.build/) rules we use for compiling, bundling, testing,
+and running our frontend code come from
+[rules_nodejs](https://github.com/bazelbuild/rules_nodejs).
+
+When upgrading rules_nodejs we generally must also consider upgrading:
+* The npm packages scoped with `@bazel`
+* The rules_sass bazel library
+* The bazel version
+
+Sample upgrade: https://github.com/tensorflow/tensorboard/pull/5977
+
+1. Determine which version of rules_nodejs to upgrade to and find its release
+   notes in https://github.com/bazelbuild/rules_nodejs/releases. Read the
+   upgrade instructions for that release and every other release since our
+   previous upgrade. Typically these instructions follow the same pattern but
+   there are occasionally special instructions, especially for major releases.
+   Make a mental note of any of these special instructions.
+
+2. Update the build_bazel_rules_nodejs target in the WORKSPACE file as described
```

Comment:
Maybe it is obvious but here they need to delete the old mirror or comment it out or something.